### PR TITLE
Remove undo related uml diagrams and screenshots. Update developer guide

### PR DESCRIPTION
**Pull request by CharlesLee01**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -240,67 +240,6 @@ Below is an example activity diagram for a valid enrol command from the user.
...
-
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
-
-![UndoRedoState4](images/UndoRedoState4.png)
-
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. Reason: It no longer makes sense to redo the `add n/David …​` command. This is the behavior that most modern desktop applications follow.
-
-![UndoRedoState5](images/UndoRedoState5.png)
 
 The following activity diagram summarizes what happens when a user executes a new command:
```
Comment (kouyk) [inline docs/DeveloperGuide.md:244]: There are leftover remaining.

Comment (CharlesLee01) [reply docs/DeveloperGuide.md:244]: Done!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -240,92 +240,11 @@ Below is an example activity diagram for a valid enrol command from the user.
...
 
 ![CommitActivityDiagram](images/CommitActivityDiagram.png)
```
Comment (kouyk) [inline docs/DeveloperGuide.md:246]: These are also part of the undo

Comment (CharlesLee01) [reply docs/DeveloperGuide.md]: OK!

**Comment by kouyk** [comment]
Some of the images that are no longer referenced are being edited and instead of removed.

**Review by kouyk — COMMENTED** [review]
There are 2 diagrams you've edited, however, I don't see the respective puml files updated. My personal advice when using git is to commit more frequently and make your changes in small increments, that way it will make it clearer for everyone to keep track of the changes and less likely to miss out on things. It need not be a commit for every single edit, but group changes in a meaningful way goes a long way. With the vague commit messages I can only guess what is really being edited and I too will miss out on some of the details and hence the many back and forth we have on this.

**Comment by CharlesLee01** [comment]
If you look properly, the addressbook in the uml diagram is changed to pawbook.

**Comment by kouyk** [comment]
> If you look properly, the addressbook in the uml diagram is changed to pawbook.

If you look properly at my previous reply, I was referring to puml files not updated while the PNG files are updated. And if you look properly at your commits, you didn't include the updated puml files :)

**Comment by CharlesLee01** [comment]
OK, I know why it happens coz our master is updated when I changed it. So when I sync my master with team's master. It is left out :(

**Comment by CharlesLee01** [comment]
Lemme do it real quick

**Comment by CharlesLee01** [comment]
Done!

File: docs/diagrams/ModelClassDiagram.puml [hunk 3]
```diff
@@ -5,11 +5,11 @@ skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
 Package Model <<Rectangle>>{
-Interface ReadOnlyAddressBook <<Interface>>
+Interface ReadOnlyPawbook <<Interface>>
```
Comment (kouyk) [inline docs/diagrams/ModelClassDiagram.puml:8]: If you look at the source code, `ReadOnlyPawbook` doesn't exist, those references to `AddressBook` has been renamed to `Database` instead.

File: docs/diagrams/ModelClassDiagram.puml [hunk 4]
```diff
@@ -5,11 +5,11 @@ skinparam arrowColor MODEL_COLOR
...
 Interface Model <<Interface>>
 Interface ObservableList <<Interface>>
-Class AddressBook
-Class ReadOnlyAddressBook
+Class Pawbook
+Class ReadOnlyPawbook
```
Comment (kouyk) [inline docs/diagrams/ModelClassDiagram.puml:12]: In case you miss out.

File: docs/diagrams/ModelClassDiagram.puml [hunk 5]
```diff
@@ -48,11 +48,11 @@ Class Tag
 Class HiddenOutside #FFFFFF
 HiddenOutside ..> Model
 
-AddressBook .up.|> ReadOnlyAddressBook
+AddressBook .up.|> ReadOnlyPawbook
```
Comment (kouyk) [inline docs/diagrams/ModelClassDiagram.puml:51]: Here's another one

File: docs/diagrams/ModelClassDiagram.puml [hunk 6]
```diff
@@ -48,11 +48,11 @@ Class Tag
...
 
 ModelManager .up.|> Model
 Model .right.> ObservableList
-ModelManager o--> "1" AddressBook
+ModelManager o--> "1" Pawbook
```
Comment (kouyk) [inline docs/diagrams/ModelClassDiagram.puml:55]: This should be the last one.

**Review by kouyk — COMMENTED** [review]
Do export the image and update it again. Also resolve your merge conflicts while you're at it! Just merge in the latest master into your current branch should work.

**Comment by kouyk** [comment]
Can you drop commit `26621e2`? Otherwise everything looks good.

**Comment by CharlesLee01** [comment]
Idk how to drop it :(. When I tried rebase in sourcetree then this happens.

**Comment by kouyk** [comment]
I don't know what you meant by rebase either, it is too vague. There are 2 types of rebase, one that changes the root commit of the branch, i.e. the point whereby the branch branched off, and another is interactive rebase, where you can change the history of the commits through a variety of options like rearranging them, squashing them aka combining, or dropping some commits, which is what you want to do. What you have done seemed like neither of them, since both will rewrite the commit history and require a force push which evidently isn't present here. From what I can tell, it seems like is you have cherry-picked an irrelevant commit from @ZhangAnli that's already in the master branch into your current branch. Give it a shot to find out how to correct your git history, what you need here is an interactive rebase, in which case you can also take the time to edit the commit that you deleted `docs/images/EditCommandScreenshot2.png` and undo the deletion such that there is no longer a merge conflict. While I can perform these operations for you, I don't want to take away the learning opportunity from you by directly modifying your branch.

**Review by kouyk — APPROVED** [review]

**Pull request merged**